Build failure on archlinux

```
[ 57%] Built target export
[ 57%] Building C object lib/darktable/plugins/lighttable/CMakeFiles/recentcollect.dir/recentcollect.c.o
[ 57%] Linking C shared module libselect.so
/home/grout/tmp/darktable/src/libs/tagging.c: In function ‘_pop_menu_dictionary_edit_tag’:
/home/grout/tmp/darktable/src/libs/tagging.c:1936:24: error: ‘private’ may be used uninitialized [-Werror=maybe-uninitialized]
 1936 |                       (gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(private)) ? DT_TF_PRIVATE : 0));
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1: all warnings being treated as errors
make[2]: *** [lib/darktable/plugins/lighttable/CMakeFiles/tagging.dir/build.make:76: lib/darktable/plugins/lighttable/CMakeFiles/tagging.dir/tagging.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3159: lib/darktable/plugins/lighttable/CMakeFiles/tagging.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 57%] Built target select
[ 57%] Linking C shared module librecentcollect.so
[ 57%] Built target recentcollect
[ 57%] Linking C shared module libcollect.so
[ 57%] Built target collect
make: *** [Makefile:156: all] Error 2
```

I built darktable successfully on March 10 (version 4.3.0+1108~gac3ec99dbc). Checking my update logs, I can't seem to find any obvious package updates as the culprit (no gcc, cmake updates). I tried to do some sort of bisection, but all of the builds I've attempted are failing. I can't seem to pinpoint what actually has changed. Has anyone else experienced this (particularly those also on archlinux)?

Platform: archlinux
GCC: 12.2.1
Built using ./build.sh script in darktable repo (with clean build directory)

--- a/src/libs/tagging.c
+++ b/src/libs/tagging.c
@@ -1809,8 +1809,8 @@ static void _pop_menu_dictionary_edit_tag(GtkWidget *menuitem, dt_lib_module_t *
   gtk_box_pack_end(GTK_BOX(box), entry, TRUE, TRUE, 0);
 
   gint flags = 0;
-  GtkWidget *category;
-  GtkWidget *private;
+  GtkWidget *category = NULL;
+  GtkWidget *private = NULL;
   GtkTextBuffer *buffer = NULL;
   if(tagid)
   {

